libs/json/types.go

```diff
@@ -0,0 +1,108 @@
+package json
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+)
+
+var (
+	// typeRegistry contains globally registered types for JSON encoding/decoding.
+	typeRegistry = newTypes()
```

Comment:
Another note: As far I remember amino put in a non-trivial and confusing amount of guarding the codec via mutexes. I highly doubt that anyone was ever using it concurrently (I mean ... why on earth would you spin up some go-routines to compete with registering your types etc 😬 ). Nonetheless, this is a change in behaviour that one _could_ mention in an upgrading doc (so users that want to switch to using this library are aware that this isn't go-routine safe anymore)?

Reply:
This is goroutine-safe, the type registry has an internal `RWMutex`. And the `typeRegistry` variable will be initialized when the package is loaded, the same package that contains `RegisterType()`. I think we're good?

The only change, I suppose, is that there is now a common mutex across all JSON encoders, while with Amino one could in principle have multiple codecs with separate mutexes. However, from the uses I've come across there's always a single codec.

Reply:
Ah, I didn't see the `sync.RWMutex`. So it isn't even sth to mention. Sorry for the noise.